rules/downgrade/src/Rector/Property/TypedPropertyRector.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Downgrade\Rector\Property;
+
+use Exception;
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PhpParser\Node\NullableType;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\UnionType;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\Downgrade\Tests\Rector\Property\TypedPropertyRector\TypedPropertyRectorTest
+ */
+final class TypedPropertyRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Changes property type definition from type definitions to `@var` annotations.', [
+            new CodeSample(
+                <<<'PHP'
+class SomeClass {
+    private string $property;
+}
+PHP
+,
+                <<<'PHP'
+class SomeClass {
+    /**
+    * @var string
+    */
+    private $property;
+}
+PHP
+
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Property::class];
+    }
+
+    /**
+     * @param Property $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->type === null) {
+            return null;
+        }
+
+        $docComment = $node->getDocComment();
+        if ($docComment !== null) {
+            $node->type = null;
+            return $node;
+        }
+
+        $node->setDocComment(new Doc(sprintf("/**\r\n* @var %s\r\n*/", $this->printType($node->type))));
+        $node->type = null;
+
+        return $node;
+    }
+
+    /**
+     * @param Identifier|Name|NullableType|UnionType $type
+     */
+    private function printType($type): string
+    {
+        if ($type instanceof NullableType) {
+            return $this->printType($type->type) . '|null';
+        }
+
+        if ($type instanceof UnionType) {
+            throw new Exception('UnionType not yet supported');
+        }
+
+        if ($type instanceof Name) {
+            $name = $type->toString();
+            if ($type->isFullyQualified()) {
+                return '\\' . $name;
+            }
+
+            return $name;
+        }
+
+        if ($type instanceof Identifier) {
+            return $type->name;
+        }
+
+        throw new Exception('Unsupported type class');
```

Comment:
There is another exception here.

This code is actually a cope-paste from https://github.com/phpstan/phpstan-src/blob/master/bin/transform-source.php